### PR TITLE
MCFE-1119: Card with Checkbox #patch

### PR DESCRIPTION
**Pull request by reedsa**
Support for a Checkbox in the Card component

[MCFE-1119](https://jira.sendgrid.net/browse/MCFE-1119)

screen shot 2018-11-30 at 1 41 07 pm

## Merge Checklist

- [X] Includes link to associated Github Issue or JIRA ticket
- [X] Has descriptive PR title and body
- [X] All changes are backwards compatible
- [X] QA'd by the SendGrid Style Guide designer
- [x] Code reviewed by a SendGrid frontend lead

### CODE REVIEW

File: stories/Card.stories.tsx [hunk 1]
```diff
@@ -108,6 +132,56 @@ stories.add('Card with Badge (Centered)', () => (
...
+          id: 'checkbox-1'
+        }}
+      >
+        <span className="small" style={{ display: 'flex', alignItems: 'center', color: '#9e9e9e' }}>5 credits</span>
+      </CardWithCheckboxExample>
+    </div>
+  </div>
+));
+
+stories.add('Card with Disabled Checkbox (Inline)', () => (
```
Comment (chrischalstrom) [inline stories/Card.stories.tsx]: need imageshots for these new stories

File: src/card.tsx [hunk 2]
```diff
@@ -23,6 +24,7 @@ export interface CardProps {
         color: ValidColor;
       }
     | React.ReactElement<any>;
+  checkbox?: CheckboxProps;
```
Comment (chrischalstrom) [inline src/card.tsx]: how about just passing in a fully blown `<Checkbox>` instead of the props of a checkbox?

File: src/card.tsx [hunk 3]
```diff
@@ -44,15 +47,32 @@ export const Card: React.SFC<CardProps> = ({
   module = false,
   ...attributes
 }) => {
+  // Add click handlers for card when checkbox is present and enabled
+  if (checkbox && !checkbox.disabled) {
```
Comment (chrischalstrom) [inline src/card.tsx]: avoid this weirdness by just using a `<Checkbox>` jsx element instead of the props of a checkbox

Comment (chrischalstrom) [reply src/card.tsx]: We could add an onClick prop to the card and pass in a `noop` for the onChange handler of the Checkbox maybe?

Comment (chrischalstrom) [reply src/card.tsx]:
```
<Card
  onClick={() => setState({ checked: !checked })}
  checkbox={(
    <Checkbox
      checked={checked}
      onClick={() => {}}
    />
  })
/>
```

Comment (reedsa) [reply src/card.tsx]: After further discussion, we're going to use the current `title` prop and have it accept either a string or a Checkbox component.

File: src/@types/react-onclickoutside.d.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+declare module 'react-onclickoutside';
```
Comment (chrischalstrom) [inline src/@types/react-onclickoutside.d.ts]: is this still needed?  don't see any reference to this module anywhere

Comment (reedsa) [reply src/@types/react-onclickoutside.d.ts]: This was broken when I ran `yarn image-snapshots`. I don't know why it may have been working otherwise...

**Comment by thinkingserious** [comment]
Hello @reedsa,<br /><br/>Thanks again for the PR!<br />                                              <br />We appreciate your contribution and look forward to                                              continued collaboration. Thanks!                                             <br /><br />Team SendGrid DX

**Pull request merged**